File: src/node/pickups/buy-pickup.js
const Easypost = require('@easypost/api');
const dotenv = require('dotenv');

dotenv.config({ path: '/Users/jhammond/git/easypost/easypost-tools/.env' });
const api = new Easypost(process.env.EASYPOST_PROD_API_KEY);

api.Pickup.retrieve('pickup_123...').then(pickup => {
    pickup.buy(
        'DHLExpress', // carrier
        'ExpressPickup' // service
    ).then(console.log).catch(console.error);
}).catch(console.log);

File: src/ruby/pickups/buy-pickup.rb
require 'date'
require 'easypost'
require 'dotenv'
require 'amazing_print'

Dotenv.load('/Users/jhammond/git/easypost/easypost-tools/.env')
EasyPost.api_key = ENV['EASYPOST_TEST_API_KEY']

pickup = EasyPost::Pickup.retrieve("pickup_b188ced4291f46ee89635e0ee13b581f")
pickup.buy(carrier: "FedEx", service: "Same-day Express Pickup")

ap pickup
